### PR TITLE
{Network} `az network public-ip ddos-protection-statu`: Deprecation pre-announcement

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

fix https://github.com/Azure/azure-cli/issues/29494

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/_breaking_change.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.core.breaking_change import register_command_group_deprecate
+
+
+register_command_group_deprecate(
+    "network public-ip ddos-protection-statu",
+    redirect="az network public-ip ddos-protection-status",
```
Comment (ReaNAiveD) [inline src/azure-cli/azure/cli/command_modules/network/_breaking_change.py:10]: Is the redirect command available in this version? If not, could the redirect message potentially confuse customers?

Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/_breaking_change.py:10]: emmm, how to pre-inform the users about the relationship between the 2 command groups then?

Comment (ReaNAiveD) [reply src/azure-cli/azure/cli/command_modules/network/_breaking_change.py:10]: Could we provide the new command group in this version? We can offer both the deprecated and recommended groups, allowing customers to transition to the new behavior smoothly. During the breaking change window, we can then remove the deprecated group.

**Review by ReaNAiveD — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**